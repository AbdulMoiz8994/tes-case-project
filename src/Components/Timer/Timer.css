.timer-container {
    background-color: var(--timer-background-color);
    height: var(--timer-height);
    width: var(--timer-width);
  }
  
  .time-display {
    height: 30%;
    font-size: 5em;
    text-align: center !important;   
    color: blue;
    font-family: 'Comfortaa', cursive;
  }
  
  .timer-button-container {
    display: flex;
    flex-direction: row;
    justify-content: center;
    height: 30%;
  }
h1{
    text-align: center;
    font-size: 3em;
    color: #2827CC;
    font-family: 'Comfortaa', cursive;
}
