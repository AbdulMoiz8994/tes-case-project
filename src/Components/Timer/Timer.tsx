import React,{useState} from 'react'




export const Timer = () => {


let [second, setSecond]=useState<number>(0)
let [minute, setMinute]=useState<number>(0)
let [hour, setHour]=useState<number>(0)
let [status, setStatus]=useState<number>(0)
let [intervals ,setIntervals]=useState<any>()

    // start button
const Run=() =>{
    if(status !== 1){
      setStatus(1)
      setIntervals(
        setInterval(() =>{

        },1000)
      )
    }
}
    // Stop button


    //Reset button


    return (
        <div>
            <h1>Timer Application</h1>
        </div>
    )
}
