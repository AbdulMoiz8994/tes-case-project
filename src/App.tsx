import React from 'react';
import './App.css';
import {Timer} from './Components/Timer/Timer'


function App() {
  return (
    <div className="app-container">
      <Timer/>
    </div>
  );
}

export default App;
