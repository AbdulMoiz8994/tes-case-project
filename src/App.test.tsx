import React from 'react'
import {shallow} from 'enzyme'
import App from './App'
import {Timer} from './Components/Timer/Timer'


describe('App',() =>{

const container=shallow(<App/>)
it('We are making Snapshot of App Comp file',() =>{
  expect(container).toMatchSnapshot()
})

it('We only checking Div',() =>{
  expect(container.find('div').length).toBe(1)
})
it('We are checking The Timer whether ava of not',() =>{
  expect(container.containsMatchingElement(<Timer/>)).toEqual(true)
})

})