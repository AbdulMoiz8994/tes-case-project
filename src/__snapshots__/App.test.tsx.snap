// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App We are making Snapshot of App Comp file 1`] = `
<div
  className="App"
>
  <Timer />
</div>
`;
